samza-core/src/main/java/org/apache/samza/coordinator/ClusterMembership.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.coordinator;
+
+/**
+ * Coordination Primitive to maintain the list of processors in the quorum
+ */
+public interface ClusterMembership {
+  /**
+   * add processor to the list
+   */
+  void registerProcessor();
```

Comment:
do we need a processor id? 
at this point, we are using just the size of the list (aka number of processors registered) and not using the id in any way.

Reply:
If the implementor is to generate the processor id then I assume we have 1-1 association between `clusterMembership` and `StreamProcessor`. I see this has certain downsides.
1.  It is an implicit requirement and not apparent from the CoordinationUtils interface standpoint which is responsible for creation of `ClusterMembership`. 
2. Having multiple instances of `ClusterMembership` in case of multiple `StreamProcessors` might not be ideal especially in terms of maintaining multiple connections to underlying system etc.

Would it extra effort or are there any downsides taking a processor id?

Reply:
added processor id.